src/status_im/ui/components/common/common.cljs

```diff
@@ -87,9 +88,15 @@
                   (assoc :accessibility-label accessibility-label))
      value]]))
 
-(defn image-contain
-  ([source] (image-contain nil source))
-  ([{:keys [style]} source]
-   [react/view {:style (merge styles/image-contain
-                              style)}
-    [react/image {:source source :resizeMode :contain :style styles/image-contain-image}]]))
+(defn image-contain [_ _]
+  (let [content-width (reagent/atom 0)]
+    (reagent/create-class
+      {:reagent-render
+       (fn [{:keys [container-style style width height]} source]
+         [react/view {:style     (merge styles/image-contain container-style)
+                      :on-layout #(reset! content-width (-> % .-nativeEvent .-layout .-width))}
```

Comment:
I tried to do a lot of things and implement adjustments to the flexbox, but without any good results :( I liked the previous function because it kind of worked, but it seems impossible to fix the two bugs described above without any additional magic, so I decided to implement a new stupidly simple one.